sw/device/silicon_creator/lib/drivers/watchdog.c

```diff
@@ -5,42 +5,59 @@
 #include "sw/device/silicon_creator/lib/drivers/watchdog.h"
 
 #include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
 
 #include "aon_timer_regs.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
 #include "pwrmgr_regs.h"
 
 enum {
   kBase = TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR,
   kPwrMgrBase = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
-  // The AON domain is always clocked at 200 Hz.
-  // TODO(lowRISC/opentitan#7385): update this after the AON frequency is
-  // formally defined in the project.
-  kAonTimerRate = 200000,
 };
 
-void watchdog_init(uint32_t timeout_ms) {
+void watchdog_init(lifecycle_state_t lc_state) {
+  // Determine the threshold cycles.
+  uint32_t threshold = 0;
+  switch (lc_state) {
```

Comment:
Would you mind hardening this switch statement?

Reply:
I had a go. I also added some redundant checks to the watchdog_configure function.

Reply:
```suggestion
  switch (launder32(lc_state)) {
```

Reply:
#10809